portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -24,9 +26,43 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
-  @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
+    private List<String> msgs;
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        msgs = new ArrayList<>();
+
+        String msg = convertToJsonUsingGson(msgs);
+        response.setContentType("application/json");
+        response.getWriter().println(msg);
+    }
+    private String convertToJsonUsingGson(List <String> msgs) {
+        Gson gson = new Gson();
+        String json = gson.toJson(msgs);
+        return json;
   }
+  /**
+   * @return the request parameter, or the default value if the parameter
+   *         was not specified by the client
+   */
+    private String getParameter(HttpServletRequest request, String name, String defaultValue) {
+        String value = request.getParameter(name);
+        if (value == null) {
+        return defaultValue;
+        }
+        return value;
+    }
+
+  @Override
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        // Get the input from the form.
+        String firstName = getParameter(request, "firstname","");
+        String lastName = getParameter(request, "lastname",""); 
+        String email = getParameter(request, "email",""); 
+        String subject = getParameter(request,"subject","");
+
+
+        // Respond with the result.
+        response.setContentType("text/html;");
+        response.getWriter().println(firstName + " " + lastName+ " " + email + " " + subject);
```

Comment:
this response is not in html. what is the resulting page supposed to look like after submitting the form?

Reply:
so far it should display the data that  obtained from the form, but my next step is to display the stored information